src/execution_result.rs

```diff
@@ -0,0 +1,60 @@
+use colored::Colorize;
+use std::{fmt, time::Duration};
+
+pub struct ExecutionResult {
```

Comment:
Nitpick: In the past, we were told not to use the term `execution` externally since it has negative connotations. Instead, we've switched to using `run` everywhere we expose this concept to partners (for examples on the Partners Dashboard). I think we should rename this struct (and associated things) to `RunResult` (or `FunctionRunResult` if you want to be more specific).

Reply:
TIL, I'll use FunctionRunResult since RunResult suggests an action of running the result which doesn't really make sense imo